pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -98,6 +100,10 @@ public abstract class FieldSpec implements Comparable<FieldSpec>, Serializable {
     }
   }
 
+  public enum MaxLengthExceedStrategy {
+    TRIM_LENGTH, FAIL_INGESTION, SUBSTITUTE_DEFAULT_VALUE, NO_ACTION
```

Comment:
Suggest replacing `FAIL_INGESTION` with `THROW_EXCEPTION`. Real-time ingestion flow is also using the transformer. When it throws exception, IIRC we will skip that row instead of failing the ingestion. IMO `THROW_EXCEPTION` can describe the behavior more accurately

Reply:
`THROW_EXCEPTION` seems more developer-friendly but not end-user friendly. It might lead to confusion around throwing errors during consumption. We can keep the names more as actions. Can we instead rename as following:
`FAIL_INGESTION` --> `SKIP_RECORD`
`NO_ACTION` --> `INGEST_ENTIRE_RECORD`

Wdyt?

Reply:
None of them can accurately describe the end-user behavior because it is a module within the ingestion pipeline, and the end-user behavior also relies on other configs (especially `continueOnError` flag). IMO we should explain the expected behavior in the documentation, and put the expected behavior of the module here to reduce confusion. `THROW_EXCEPTION` might be a little bit java specific, maybe `ERROR`?

Reply:
`continueOnError` doesn't affect this Sanitization flow at all right? Even if continueOnError is true, we would still end up skipping the record if we throw exception from SanitizationTransformer.

```
try {
          _recordEnricherPipeline.run(decodedRow.getResult());
          _transformPipeline.processRow(decodedRow.getResult(), reusedResult);
        } catch (Exception e) {
          _numRowsErrored++;
          // when exception happens we prefer abandoning the whole batch and not partially indexing some rows
          reusedResult.getTransformedRows().clear();
          String errorMessage =
              String.format("Caught exception while transforming the record at offset: %s , row: %s", offset,
                  decodedRow.getResult());
          _segmentLogger.error(errorMessage, e);
          _realtimeTableDataManager.addSegmentError(_segmentNameStr, new SegmentErrorInfo(now(), errorMessage, e));
        }
```

Reply:
Updated to `ERROR` based on discussion offline.